Sync v3.1.0 qol

Sneh1999 opened the pull request:
Creating PR for v3.1.0-qo1. This is the same version that is supported for migration and we should use in v3.6.7

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1210675058805145

gvladika committed:
Do not exclude batch reports when using fee token

gvladika committed:
Add condition for submitting the batch report, and do the scaling

gvladika committed:
Add IFeeTokenPricer

gvladika committed:
Add getter and setter for pricer to the interface

gvladika committed:
Add setter for feeTokenPricer

gvladika committed:
Add missing param

gvladika committed:
Use new SeqInbox init param

gvladika committed:
Make existing tests work with feeTokenPricer

gvladika committed:
Format

gvladika committed:
Add setFeeTokenPricer tests

yahgwai committed:
Fee token pricing deployment

yahgwai committed:
Merge branch 'custom-fee-rollup' of https://github.com/OffchainLabs/nitro-contracts into custom-fee-rollup

gvladika committed:
Revert if trying to set fee token pricer when fee token is not used

gvladika committed:
Add fuzz test for different exchange rates

gvladika committed:
Handle overflow case due to too high exchange rate

yahgwai committed:
Updated erc20rollupeventinbox to do gas price scaling

yahgwai committed:
Merge branch 'custom-fee-rollup' of https://github.com/OffchainLabs/nitro-contracts into custom-fee-rollup

yahgwai committed:
Added integration test for batch poster reimbursement

gvladika committed:
Update rollupInitialized test

gvladika committed:
Refactor tests and check initial cost is properly reported

yahgwai committed:
Merge branch 'custom-fee-rollup' of https://github.com/OffchainLabs/nitro-contracts into custom-fee-rollup

yahgwai committed:
Added fee token pricer arg

yahgwai committed:
Use token fee pricer testnode ref

yahgwai committed:
Updated int fee token tests

yahgwai committed:
Formatting

yahgwai committed:
Updated comments

yahgwai committed:
Formatting

yahgwai committed:
Try tests with init

yahgwai committed:
Added separate job for e2e tests

yahgwai committed:
Missed test name change

yahgwai committed:
Disable while true condition linting

yahgwai committed:
Fixed 4844 tests

yahgwai committed:
Added fee token pricer address zero to arb rollup spec

yahgwai committed:
Formatting

yahgwai committed:
Updated orbit ame

yahgwai committed:
Updated sigs and storage

yahgwai committed:
Added estimate gas

yahgwai committed:
L1 wal balance console

yahgwai committed:
Console logging

yahgwai committed:
Added gas price estimation

gvladika committed:
Update Slither db, no new findings

gvladika committed:
Add fuzz test for rollupInitialized

gvladika committed:
Add ConstantExchangeRatePricer

gvladika committed:
Add OwnerAdjustableExchangeRatePricer

gvladika committed:
Add uniswap v2 dependency

gvladika committed:
Add uniswap lib

gvladika committed:
Uniswap v2 twap draft

gzeoneth committed:
Merge branch 'develop' into custom-fee-rollup

gvladika committed:
Use local math files

gvladika committed:
Return exchange rate

gvladika committed:
AMM trade tracker draft

gvladika committed:
Return exchange rate

gvladika committed:
Add comments

gvladika committed:
Fetch tokens

gvladika committed:
Add gas reporting hook

gvladika committed:
Update internal state

gvladika committed:
Track exchange rate per spender

gvladika committed:
No need for ownable

gvladika committed:
Avoid underflow

gvladika committed:
Owner sets deafult exchange rate

gvladika committed:
Prvoide min amount of eth received to protect from slippage

gvladika committed:
Scale to 18 decimals

gvladika committed:
Use safe erc20

gvladika committed:
Test swapTokenToEth

gvladika committed:
Test getExchangeRate

gvladika committed:
Fix >18 decimals scaling

gvladika committed:
Simplify token spending calculation

gvladika committed:
Test state updates after batch is posted

gvladika committed:
Make calldata cost adjustable

gvladika committed:
No updates if ethAcc is 0

gvladika committed:
Control who can call gas hook and update internal state

gvladika committed:
Add events and custom error

gvladika committed:
Merge branch 'develop' into custom-fee-rollup

gvladika committed:
Clean up

gvladika committed:
Update slither db

gvladika committed:
Update audit-ci.jsonc

gvladika committed:
Provide trade deadline from outside

gvladika committed:
Move fork tests to separate file

gvladika committed:
Merge branch 'custom-fee-rollup' into fee-token-pricers

yahgwai committed:
Updated format command

yahgwai committed:
Comment updates

yahgwai committed:
Merge from develop

yahgwai committed:
Updated slither

yahgwai committed:
Updated slither

yahgwai committed:
Merge from bold-merge

yahgwai committed:
Formatting

yahgwai committed:
Updated signatures and storage

yahgwai committed:
Slither

yahgwai committed:
Merge from custom-fee-rollup

yahgwai committed:
Toml update

yahgwai committed:
Formatting and uniswap install

yahgwai committed:
Added trade tracker instead of amm trade tracker, and a readme

yahgwai committed:
Updated readme and comments

yahgwai committed:
Merge from develop

yahgwai committed:
Formatting

yahgwai committed:
Merge branch 'custom-fee-rollup' into fee-token-pricers

yahgwai committed:
Formatting

gzeoneth committed:
Merge pull request #288 from OffchainLabs/develop

release: v3.0.0

gzeoneth committed:
chore: EdgeChallengeManager metadata

gzeoneth committed:
chore: add rei meta

yahgwai committed:
Merge branch 'develop' into custom-fee-rollup

yahgwai committed:
Updated fee token pricer zero in test helpers

yahgwai committed:
Updated force include spec

gzeoneth committed:
Merge branch 'custom-fee-rollup' into fee-token-pricers

yahgwai committed:
Test timeout action testnode

yahgwai committed:
Dummy commit

yahgwai committed:
First commit at updated cargo stylus factory

yahgwai committed:
Fixed seq inbox tests

yahgwai committed:
Renamed stylus deployer

yahgwai committed:
Added tests for stylus deployer

yahgwai committed:
Cleaned TODOs

yahgwai committed:
Merge branch 'main' into stylus-factory

yahgwai committed:
Merge from stylus factory

yahgwai committed:
Updated format

yahgwai committed:
Use hardhat config solc pragma

yahgwai committed:
Use stable foundry

yahgwai committed:
Slither update

yahgwai committed:
Whitespace change

yahgwai committed:
Merge branch 'custom-fee-rollup' into fee-token-pricers

yahgwai committed:
Use stable foundry

yahgwai committed:
Use main testnode

yahgwai committed:
Update src/stylus/StylusDeployer.sol

Co-authored-by: Henry <11198460+godzillaba@users.noreply.github.com>

yahgwai committed:
Update src/stylus/StylusDeployer.sol

Co-authored-by: Henry <11198460+godzillaba@users.noreply.github.com>

yahgwai committed:
Updates from code review

yahgwai committed:
Merge branch 'stylus-factory-chris' of https://github.com/OffchainLabs/nitro-contracts into stylus-factory-chris

yahgwai committed:
Updated format

yahgwai committed:
Updated slither

yahgwai committed:
Updated assembly block

yahgwai committed:
Formattinh

godzillaba committed:
deploy erc20 impls

yahgwai committed:
Added pos arg

yahgwai committed:
Test ci

godzillaba committed:
works with fresh arbsepoli rollup

godzillaba committed:
fix execution script

godzillaba committed:
remove addresses from custom config

godzillaba committed:
remove deployed contracts

yahgwai committed:
Try known arg combination

yahgwai committed:
Try different startup

yahgwai committed:
Added pos arg

yahgwai committed:
Added no wait for l3 token bridge

yahgwai committed:
Added testnode ref

yahgwai committed:
Removed reference to fee token pricer branch

yahgwai committed:
Try fee token pricer ref branch

yahgwai committed:
Added fee token pricer args

godzillaba committed:
Use stable foundry (#298)

Co-authored-by: Chris Buckland <cpbuckland88@gmail.com>

diegoximenes committed:
ArbOS11To32UpgradeTest develop base branch (#286)

* ArbOS11To32UpgradeTest

* Uses mcopy in ArbOs11TO32UpgradeTest

* Fix lint issues

---------

Co-authored-by: gzeon <hng@offchainlabs.com>

gzeoneth committed:
Merge branch 'develop' into stylus-factory-chris

eljobe committed:
Add a BigMap contract for testing the storage trie (#295)

Co-authored-by: gzeon <im@gzeon.dev>

gzeoneth committed:
Merge branch 'develop' into bigmap-develop

gzeoneth committed:
Merge pull request #299 from OffchainLabs/bigmap-develop

chore: bigmap mock and update metadata hash script

gzeoneth committed:
Merge branch 'develop' into stylus-factory-chris

gzeoneth committed:
v3.0.1-beta.0

gzeoneth committed:
Merge pull request #294 from OffchainLabs/stylus-factory-chris

Stylus deployer

yahgwai committed:
Use named branch

godzillaba committed:
Merge branch 'develop' into custom-fee-bold-script

yahgwai committed:
Comment commit

godzillaba committed:
test - nocache in foundry install

yahgwai committed:
Added rollup creation changes

godzillaba committed:
Merge branch 'develop' into custom-fee-rollup-ci

yahgwai committed:
Merge pull request #302 from OffchainLabs/custom-fee-rollup-ci

Custom fee rollup ci

yahgwai committed:
Test output

yahgwai committed:
Merge branch 'custom-fee-rollup' of https://github.com/OffchainLabs/nitro-contracts into custom-fee-rollup

yahgwai committed:
Try timeout testnode

yahgwai committed:
Add non zero loser stake escrow

yahgwai committed:
Added todo

yahgwai committed:
Comment updates

yahgwai committed:
Merge branch 'custom-fee-rollup' into fee-token-pricers

yahgwai committed:
Formatting

yahgwai committed:
Formatting

yahgwai committed:
Merge branch 'custom-fee-rollup' into fee-token-pricers

eljobe committed:
Add methods to enable/disable and check the EIP-7623 feature (#311)

* Add methods to enable/disable and check the EIP-7623 feature

This precompile will allows the feature to be off by default, but toggled on and
off via a single call to the ArbOwner contract.

Part of: NIT-3139

* format: yarn format

* docs: note applicable arbos version

* chore: move getter to ArbOwnerPublic

* chore: mark getter as view

---------

Co-authored-by: gzeon <im@gzeon.dev>

eljobe committed:
Adds the setWasmMaxSize precompile function to ArbOwner

Chain owners that wish to allow larger wasm contracts can increase the size via
an ArbOwner call on their chain.

Part of: NIT-3173

eljobe committed:
Make a tabs-to-spaces pass

eljobe committed:
Format the contract.

godzillaba committed:
Merge branch 'develop' into custom-fee-bold-script

godzillaba committed:
disable delay buffer by default

gzeoneth committed:
chore: bump dependencies (#308)

* fix: remove unused dependency

* chore: bump dependencies

* chore: remove from audit ci whitelist

* chore: disable bold upgrade test

* ci: use foundry stable

* chore: bump dependencies

* test: use new revertedWithCustomError

* chore: bump more dependencies

* chore: remove issue from whitelist

* fix: pretty is now default

* fix: allow_internal_expect_revert for lib test

* chore: whitelist advisory

* chore: bump minors in package.json

* revert: non-dev dependencies bump

* test - nocache in foundry install (#301)

Co-authored-by: gzeon <hng@offchainlabs.com>

* test: fix

---------

Co-authored-by: Henry <11198460+godzillaba@users.noreply.github.com>

godzillaba committed:
Merge branch 'develop' into custom-fee-bold-script

godzillaba committed:
disable validator whitelist by default

godzillaba committed:
Revert "disable validator whitelist by default"

This reverts commit 8d07184771e0439fd3c0ee1ddfd8397a3162f386.

godzillaba committed:
ENABLE validator whitelist by default

godzillaba committed:
remove testnode validator

godzillaba committed:
update defaults

godzillaba committed:
custom is default config

godzillaba committed:
fix buffer config

godzillaba committed:
additional erc20 specific checks in bold upgrade script

godzillaba committed:
fix comment

eljobe committed:
Merge branch 'develop' into set-max-wasm

eljobe committed:
Merge pull request #316 from OffchainLabs/set-max-wasm

Adds the setWasmMaxSize precompile function to ArbOwner

eljobe committed:
Multi redeem public (#313)

* Add a method for redeeming a list of retryable ticket ids

This will allow us to test that when many retryables are redeemed in the same
block, that the execution client doesn't have any problems.

Related to: NIT-3010

* Add the gas cost to the call

* Add the ability to create addresses to the RedeemAll method

This change also renames the method to ReedeemAllAndCreateAddresses. The reason
for the complexity of the commit is that we need to have a single top-level
contract call that will cause many transactions to occur in the same block. Some
of those transactions should cause accounts to be deleted, and others should
cause accounts to be created.

This way, we can tickle a bug in the current implementation of validation which
occurs when account deletions and additions take place out of order and the
validation logic isn't able to find a preimage for a particular account.

* Convert tabs to spaces

* format: yarn format

---------

Co-authored-by: gzeon <im@gzeon.dev>
Co-authored-by: gzeon <hng@offchainlabs.com>

godzillaba committed:
add custom hardhat network for bold scripts (#317)

* add custom hardhat network

* fix config

* remove l1 rpc from env

gzeoneth committed:
Merge branch 'develop' into custom-fee-bold-script

gzeoneth committed:
fix: hardhat deploy getContract is deprecated

gzeoneth committed:
fix: unused

gzeoneth committed:
fix: test edge case

gzeoneth committed:
Merge pull request #297 from OffchainLabs/custom-fee-bold-script

Update BOLD prepare script for orbit

amsanghi committed:
SelfDestruct helper contracts for unit testing SelfDestruct (#319)

* SelfDestruct helper contracts for unit testing SelfDestruct

* fix format

---------

Co-authored-by: gzeon <hng@offchainlabs.com>

gzeoneth committed:
Merge branch 'develop' into fix-hh-deploy

gzeoneth committed:
fix: v3 rollup deployment tool (#312)

* Updated config

* feat: add stake-token

* chore: log weth deploy

* chore: wait deployed

* fix: deploy weth with deployerWallet

* workaround: use smaller stake size for test

* workaround: add testnode validator signer as validator

* fix: remove unused dependency

* chore: bump dependencies

* chore: remove from audit ci whitelist

* chore: disable bold upgrade test

* ci: use foundry stable

* chore: bump dependencies

* test: use new revertedWithCustomError

* chore: bump more dependencies

* chore: remove issue from whitelist

* fix: pretty is now default

* fix: allow_internal_expect_revert for lib test

* chore: whitelist advisory

* chore: bump minors in package.json

* revert: non-dev dependencies bump

* test - nocache in foundry install (#301)

Co-authored-by: gzeon <hng@offchainlabs.com>

* fix: genesis machine finished

* try realistic params

* lower min assertion period

---------

Co-authored-by: Chris Buckland <cpbuckland88@gmail.com>
Co-authored-by: Henry <11198460+godzillaba@users.noreply.github.com>

gzeoneth committed:
Merge branch 'develop' into fix-hh-deploy

gzeoneth committed:
Merge pull request #281 from OffchainLabs/fee-token-pricers

Fee token pricers

gzeoneth committed:
Merge remote-tracking branch 'origin/develop' into custom-fee-rollup

gzeoneth committed:
ci: update testnode ref

gzeoneth committed:
chore: fix 4b and storage

gzeoneth committed:
fix: skip custom fee token test for eth rollup

gzeoneth committed:
fix: use delay proof sig

gzeoneth committed:
fix: isbold true

gzeoneth committed:
fix: stylus test

gzeoneth committed:
chore: update RollupCreated event sig

gzeoneth committed:
chore: speed up confirmation

gzeoneth committed:
fix: dummy stake token

gzeoneth committed:
test: workaround

gzeoneth committed:
ci: run all e2e

gzeoneth committed:
ci: skip test if the fee token pricer is not set

gzeoneth committed:
ci: dont wait forever

gzeoneth committed:
Merge remote-tracking branch 'origin/fix-hh-deploy' into custom-fee-rollup

gzeoneth committed:
fix: hardhat deploy getContract is deprecated (#322)

* fix: hardhat deploy getContract is deprecated

* fix: unused

* fix: test edge case

godzillaba committed:
Merge branch 'develop' into custom-fee-rollup

gzeoneth committed:
docs: remove stale comment

gzeoneth committed:
chore: remove unused import

gzeoneth committed:
format: yarn format

gzeoneth committed:
ci: enable e2e

gzeoneth committed:
ci: fix e2e

gzeoneth committed:
revert: #294

gzeoneth committed:
Merge branch 'revert-pr-294' into fix-format

gzeoneth committed:
feat: FeeTokenPricerSet event (#324)

gzeoneth committed:
chore: bump sdk to 3.7.3

gzeoneth committed:
release: v3.0.1

gzeoneth committed:
Merge pull request #326 from OffchainLabs/fix-format

fix: e2e test and format

gzeoneth committed:
Merge remote-tracking branch 'origin/develop' into custom-fee-rollup

gzeoneth committed:
chore: remove patch

gzeoneth committed:
chore: update slither db

gzeoneth committed:
release: v3.1.0

gzeoneth committed:
Merge pull request #252 from OffchainLabs/custom-fee-rollup

Support for rollup mode when using fee token

gzeoneth committed:
fix: deploy scripts (#329)

* chore: also verify with sourcify

* chore: update example

* fix: better typing

* ci: bump action versions

* test: workaround delay proof

* format: yarn format

* ci: fix slither db using new version

TucksonDev committed:
Add vuln to audit-ci allowlist

TucksonDev committed:
Merge pull request #342 from OffchainLabs/audit-ci-fix

Add vuln to audit-ci allowlist

TucksonDev committed:
Bold upgrade action QoL improvements (#333)

* Simplify code

* Format

* Add new RollupCreators

* Improve starting validation and final verification

* Use delayBufferable SequencerInbox and hardcode templates

* Format

* Update testnode config

* Update testnode config II

* Add MAX_DATA_SIZE from environment if present (for testnode)

* Add MAX_DATA_SIZE from environment if present (for testnode) II

* Test changes

* Test changes II

* Test changes III

* Add optional deployer key from env variable

* Add nitro-testnode templates

* Format

* Validate that stakeAmt is not 0

* Add function to verify creator templates

---------

Co-authored-by: gzeon <hng@offchainlabs.com>

lukeiannucci committed:
merge

Sneh1999 committed:
initial commit

Sneh1999 committed:
small fixes to remove via-ir

Sneh1999 committed:
fix all tests

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EspressoSystems/nitro-contracts?pullRequest=84) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/EspressoSystems/nitro-contracts?pullRequest=84) before we can accept your contribution.<br/>**2** out of **5** committers have signed the CLA.<br/><br/>:white_check_mark: lukeiannucci<br/>:white_check_mark: Sneh1999<br/>:x: godzillaba<br/>:x: TucksonDev<br/>:x: gzeoneth<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EspressoSystems/nitro-contracts?pullRequest=84) it.</sub>

Sneh1999 committed:
merge

Sneh1999 committed:
merge

Sneh1999 committed:
rerun ci

Sneh1999 committed:
fix ci

Sneh1999 committed:
fix tests

Sneh1999 committed:
revert branch to original

ImJeremyHe commented on test/contract/sequencerInboxDelayBufferable.spec.ts:
```diff
@@ -245,6 +246,23 @@ describe('SequencerInboxDelayBufferable', async () => {
 
     const tx = sequencerInbox
       .connect(batchPoster)
+<<<<<<< HEAD
```
Unhandled conflict

Sneh1999 replied:
Thanks @ImJeremyHe, I updated it https://github.com/EspressoSystems/nitro-contracts/pull/84/commits/b4c167e4001d6e1a211a769e33c3dd4a951ac216

ImJeremyHe reviewed (commented):
I reviewed this PR by reviewing the comparision with upstream 3.1.0

https://github.com/OffchainLabs/nitro-contracts/compare/v3.1.0...EspressoSystems:nitro-contracts:sync-v3.1.0-qol

I think this PR is great and it seems to handle most of the conflicts correctly.

Despite of those files formatted on our side(though it is not a good practice formatting upstreams' files) and those changes I can understand, we have these files different with upstream:
- src/bridge/ERC20Bridge.sol
- scripts/executeBoldUpgrade.ts
- scripts/deploymentUtils.ts
- scripts/boldUpgradeFunctions.ts (we removed a lot of lines)
- scripts/boldUpgradeCommon.ts

Maybe there are more.

These differences are not soly coming from this PR though, I would suggest we can keep them same.

Sneh1999 committed:
fix based on comments

Sneh1999 commented:
> I reviewed this PR by reviewing the comparision with upstream 3.1.0
> 
> [OffchainLabs/nitro-contracts@v3.1.0...EspressoSystems:nitro-contracts:sync-v3.1.0-qol](https://github.com/OffchainLabs/nitro-contracts/compare/v3.1.0...EspressoSystems:nitro-contracts:sync-v3.1.0-qol)
> 
> I think this PR is great and it seems to handle most of the conflicts correctly.
> 
> Despite of those files formatted on our side(though it is not a good practice formatting upstreams' files) and those changes I can understand, we have these files different with upstream:
> 
> * src/bridge/ERC20Bridge.sol
> * scripts/executeBoldUpgrade.ts
> * scripts/deploymentUtils.ts
> * scripts/boldUpgradeFunctions.ts (we removed a lot of lines)
> * scripts/boldUpgradeCommon.ts
> 
> Maybe there are more.
> 
> These differences are not soly coming from this PR though, I would suggest we can keep them same.

@ImJeremyHe Apologies, I forgot to mention that we need to compare with branch `3.1.0-qol` because this is also the branch that will be used for migration as mentioned in the docs [here](https://github.com/OffchainLabs/orbit-actions/tree/main/scripts/foundry/contract-upgrades/3.1.0#how-to-use-it)

However you are right, I did miss some changes:
I updated `src/bridge/ERC20Bridge.sol`

Other files I looked at their is no diff or the changes are important for our integration. Please let me know if you have any further questions!

jjeangal commented on .gitmodules:
```diff
@@ -1,3 +1,8 @@
+[submodule "lib/forge-std"]
```
Not sure if this is doable or viable but optimally we could also use `forge-std` which is already imported in `lib/espresso-tee-contracts` to avoid redundancy. When I tried in another branch it was able to build and run successfully.

Sneh1999 replied:
I am worried that arbitrum labs would want a different version of forge std than what we will have in lib/espresso-tee-contracts ( probably they will support an older version)

jjeangal commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "@espressosystems/nitro-contracts",
-  "version": "3.0.0",
+  "name": "@arbitrum/nitro-contracts",
```
Not @espressosystems ?

jjeangal commented on test/challengeV2/StateTools.sol:
```diff
@@ -4,37 +4,46 @@
-import "../../src/state/Machine.sol";
-import "../../src/rollup/RollupLib.sol";
-import "./Utils.sol";
+import 'forge-std/Test.sol';
```
Nit: Should switch to ( " ) instead of ( ' ) for repository format guidelines.

jjeangal commented on test/contract/sequencerInboxDelayBufferable.spec.ts:
```diff
@@ -477,19 +481,21 @@ describe('SequencerInboxDelayBufferable', async () => {
-      '0x',
-      { gasLimit: 10000000 }
+      [
+        'addSequencerL2BatchFromOrigin(uint256,bytes,uint256,address,uint256,uint256)'
```
Why are we not calling the espresso function here? Is it because we are not enabling delay proofs?

Sneh1999 replied:
True, I had to add it. Thanks!

jjeangal commented on test/contract/sequencerInboxForceInclude.spec.ts:
```diff
@@ -430,25 +426,23 @@ describe('SequencerInboxForceInclude', async () => {
-      espressoMetadata
-    )
+      [
+        'addSequencerL2BatchFromOrigin(uint256,bytes,uint256,address,uint256,uint256)'
```
Same here

Sneh1999 replied:
Thanks for catching this!

jjeangal commented on test/foundry/SequencerInboxTEE.t.sol:
```diff
@@ -1,458 +1,520 @@
-import {IEspressoTEEVerifier} from "espresso-tee-contracts/interface/IEspressoTEEVerifier.sol";
-import {EspressoNitroTEEVerifier} from "espresso-tee-contracts/EspressoNitroTEEVerifier.sol";
-import {CertManager} from "@nitro-validator/CertManager.sol";
+import 'forge-std/Test.sol';
```
Nit: Should also be " "

jjeangal commented on test/contract/sequencerInbox.spec.4844.ts:
```diff
@@ -50,7 +50,7 @@ import {
 import { Toolkit4844 } from './toolkit4844'
```
Should probably also be " " :face_with_spiral_eyes:

Sneh1999 replied:
Yea seems like they are using diff things for solidity and typescript

Sneh1999 committed:
address comments

ImJeremyHe commented:
Great! Thanks @Sneh1999
